### PR TITLE
Minor changes and reductions of include dependencies

### PR DESCRIPTION
**Pull request by markuspg**

### CODE REVIEW

File: libuuu/tar.cpp [hunk 1]
```diff
@@ -14,7 +14,7 @@
 #include <iostream>
 using namespace std;
 
-int Tar::Open(string filename)
+int Tar::Open(const std::string &filename)
 {
```
Comment (nxpfrankli) [inline libuuu/tar.cpp:18]: Already using name space std.

std:: is very noise.

File: libuuu/buffer.cpp [hunk 2]
```diff
@@ -59,7 +59,7 @@ static mutex g_mutex_map;
 
 string g_current_dir = ">";
 
-void set_current_dir(string dir)
+void set_current_dir(const std::string &dir)
 {
```
Comment (nxpfrankli) [inline libuuu/buffer.cpp:63]: in Cpp file, avoid use std::

**Comment by markuspg** [comment]
Hi nxpfrankli,

thank you for the clarification. I added "using namespace std" back to the the respective files.

**Comment by nxpfrankli** [comment]
Please fix build problem at commit which caused.  Acutally, we should avoid build error as much as possible. build error will block git-bisect in future.

**Comment by nxpfrankli** [comment]
Please fix windows build error

**Pull request merged**